### PR TITLE
Adds 9bit reception support to the TI uart drivers.

### PR DESCRIPTION
**Pull request by balazsracz**
Implements a 9-bit reception mode for Tiva and CC32xx UART drivers.
The nine-bit reception works by reading two byte payloads from the fd for each incoming byte, LSB-first. The second byte contains the parity bit. The special mode is enabled with an ioctl, and only works for RX, not for TX.

### CODE REVIEW

File: include/freertos/tc_ioctl.h [hunk 1]
```diff
@@ -46,11 +46,14 @@
...
 #define TCPARONE    IO(TERMIOS_IOC_MAGIC, 0xF3)
 #define TCPARZERO   IO(TERMIOS_IOC_MAGIC, 0xF4)
+/// Use 9-bit reception mode
+#define TCNINEBITRX IO(TERMIOS_IOC_MAGIC, 0xF5)
 /// One stop bit
 #define TCSTOPONE   IO(TERMIOS_IOC_MAGIC, 0xF8) 
 /// Two stop bits
 #define TCSTOPTWO   IO(TERMIOS_IOC_MAGIC, 0xF9)
 
+
```
Comment (bakerstu) [inline include/freertos/tc_ioctl.h:56]: Extra whitespace or intentional?

File: src/freertos_drivers/ti/CC32xxUart.cxx [hunk 2]
```diff
@@ -289,14 +297,34 @@ void CC32xxUart::interrupt_handler()
...
-            rxBuf->signal_condition_from_isr();
+            else
+            {
+                // parity error bit is moved to the ninth bit, then two bytes
+                // are written to the buffer.
+                long bit9 = (data & 0x200) >> 1;
+                data &= 0xff;
+                data |= bit9;
+                rxBuf->put((uint8_t *)&data, 2);
+                rxBuf->signal_condition_from_isr();
```
Comment (bakerstu) [inline src/freertos_drivers/ti/CC32xxUart.cxx:314]: The if/else [overrun] logic for signal_condition_from_isr() is not the same for both 9 and 8 bit data paths. I think technically either is actually fine, but we should be consistent. Pick one and modify the other to match.

File: src/freertos_drivers/ti/TivaUart.cxx [hunk 3]
```diff
@@ -218,14 +219,34 @@ void TivaUart::interrupt_handler()
...
+            }
+            else
+            {
+                // parity error bit is moved to the ninth bit, then two bytes
+                // are written to the buffer.
+                long bit9 = (data & 0x200) >> 1;
+                data &= 0xff;
+                data |= bit9;
+                rxBuf->put((uint8_t *)&data, 2);
+                rxBuf->signal_condition_from_isr();
```
Comment (bakerstu) [inline src/freertos_drivers/ti/TivaUart.cxx:236]: Same here.

**Review by bakerstu — APPROVED** [review]

File: src/freertos_drivers/ti/CC32xxUart.cxx (hunk 2 above)
Comment (balazsracz) [reply src/freertos_drivers/ti/CC32xxUart.cxx:314]: fixed

File: include/freertos/tc_ioctl.h (hunk 1 above)
Comment (balazsracz) [reply include/freertos/tc_ioctl.h]: fixed

File: src/freertos_drivers/ti/TivaUart.cxx (hunk 3 above)
Comment (balazsracz) [reply src/freertos_drivers/ti/TivaUart.cxx:236]: fixed

**Pull request merged**